org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -16,63 +16,106 @@ namespace MixedReality.Toolkit.Input
+
+        /// <summary>
+        /// Used to keep track of the previously detected interactrables so that we can know which
+        /// interactable stopped being detected and update their buttons front plate RawImage.
+        /// </summary>
+        private readonly HashSet<IXRProximityInteractable> previouslyDetectedInteractables = new();
 
         /// <summary>
-        /// Used to keep track of the previously detected colliders so that we can know which
-        /// colliders stopped being detected and update their buttons front plate and rounded
-        /// rect if they are labeled as dynamic (based on proximity).
+        /// Stores the currently detected interactables (updated every frame).
         /// </summary>
-        List<Collider> previouslyDetectedColliders = new List<Collider>();
+        private readonly List<IXRProximityInteractable> currentlyDetectedInteractables = new();
+
+        /// <summary>
+        /// Stores the interactables that are no longer detected (updated every frame).
+        /// </summary>
+        private readonly List<IXRProximityInteractable> noLongerDetectedInteractables = new();
 
         /// <inheritdoc />
         public override bool IsModeDetected()
         {
             bool result = base.IsModeDetected() || IsNearInteractorSelecting();
             if (result)
             {
-                for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+                UpdateCurrentlyDetectedInteractables();
+                UpdateProximityExited();
+                UpdateProximityEntered();
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Calls OnProximityExited for all interactables that were previously ProximityHover detected but are no longer detected.
+        /// </summary>
+        private void UpdateProximityExited()
+        {
+            noLongerDetectedInteractables.Clear();
+
+            foreach (IXRProximityInteractable previouslyDetectedInteractable in previouslyDetectedInteractables)
+            {
+                if (!currentlyDetectedInteractables.Contains(previouslyDetectedInteractable))
+                {
+                    noLongerDetectedInteractables.Add(previouslyDetectedInteractable);
+                }
+            }
+
+            foreach (IXRProximityInteractable noLongerDetectedInteractable in noLongerDetectedInteractables)
+            {
+                noLongerDetectedInteractable.OnProximityExited(new ProximityHoverExitedEventArgs(noLongerDetectedInteractable));
+                currentlyDetectedInteractables.Remove(noLongerDetectedInteractable);
+            }
+        }
+
+        /// <summary>
+        /// Call OnProximityEntered for all interactables that are currently detected but were not detected previously.
+        /// </summary>
+        private void UpdateProximityEntered()
+        {
+            foreach (IXRProximityInteractable currentlyDetectedInteracable in currentlyDetectedInteractables)
```

Comment:
nit: typo in `currentlyDetectedInteracable` (missing the final `t` in `Interactable`)

Reply:
Fixed in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/659/commits/996c8069b8dddc09d3aeaf7977f2114842cfd025, resolving.